lib/pliny/errors.rb

```diff
@@ -1,117 +1,64 @@
+require "i18n"
+
 module Pliny
   module Errors
     class Error < StandardError
-      attr_accessor :id
 
-      def self.render(error)
-        headers = { "Content-Type" => "application/json; charset=utf-8" }
-        data = { id: error.id, message: error.message }
-        [error.status, headers, [MultiJson.encode(data)]]
-      end
+      class << self
+        attr_accessor :error_class_id, :error_class_status
 
-      def initialize(message, id)
-        @id = id
-        super(message)
+        def render(error)
+          headers = { "Content-Type" => "application/json; charset=utf-8" }
+          data = { id: error.id, message: error.user_message }.merge(error.metadata)
+          [error.status, headers, [MultiJson.encode(data)]]
+        end
       end
-    end
 
-    class HTTPStatusError < Error
-      attr_accessor :status
+      attr_accessor :id, :status, :metadata
 
-      def initialize(message = nil, id = nil, status = nil)
-        meta    = Pliny::Errors::META[self.class]
-        message = message || meta[1] + "."
-        id      = id || meta[1].downcase.tr(' ', '_').to_sym
-        @status = status || meta[0]
-        super(message, id)
+      def initialize(id=nil, metadata: {})
+        @id = (id || self.class.error_class_id).to_sym
+        @status = self.class.error_class_status
+        @metadata = metadata
+        super(@id.to_s)
+      end
+
+      def user_message
+        I18n.t("errors.#{self.id}")
       end
     end
 
-    class Continue < HTTPStatusError; end                     # 100
-    class SwitchingProtocols < HTTPStatusError; end           # 101
-    class OK < HTTPStatusError; end                           # 200
-    class Created < HTTPStatusError; end                      # 201
-    class Accepted < HTTPStatusError; end                     # 202
-    class NonAuthoritativeInformation < HTTPStatusError; end  # 203
-    class NoContent < HTTPStatusError; end                    # 204
-    class ResetContent < HTTPStatusError; end                 # 205
-    class PartialContent < HTTPStatusError; end               # 206
-    class MultipleChoices < HTTPStatusError; end              # 300
-    class MovedPermanently < HTTPStatusError; end             # 301
-    class Found < HTTPStatusError; end                        # 302
-    class SeeOther < HTTPStatusError; end                     # 303
-    class NotModified < HTTPStatusError; end                  # 304
-    class UseProxy < HTTPStatusError; end                     # 305
-    class TemporaryRedirect < HTTPStatusError; end            # 307
```

Comment:
I know that removing all the non 4xx-5xx error classes is going to cause things to break in a few of my apps. To make matters a little worse, I suspect a few of them might be used for control flow.

That said, I do think this is a good change and what I've been doing is BAD (tm).